include/broker/zeek.hh

```diff
@@ -105,6 +130,104 @@ public:
     return get<std::string>(get<vector>(as_vector()[2])[0]);
   }
 
+  /// Support iteration with structured binding.
+  class metadata_iterator {
+    using iterator_category = std::forward_iterator_tag;
+    using difference_type = std::ptrdiff_t;
+    using value_type = std::pair<count, const data&>;
+    using pointer = value_type*;
+    using reference = value_type&;
+
+  public:
+    metadata_iterator(const broker::data* ptr) : ptr_(ptr) {}
+
+    const std::pair<count, const data&> operator*() const {
```

Comment:
I don't know what you're intention was to add a `const` to the return type. If it wasn't a mistake: it's not doing what you think it does. 😉

Also, don't repeat yourself:

```suggestion
    value_type operator*() const {
```

Reply:
> If it wasn't a mistake: it's not doing what you think it does. wink

Thanks for pointing this out. It's not the first time. I'll try to remember.